src/Lexer.js

```diff
@@ -67,33 +62,30 @@ const commentRegex = new RegExp(`^${commentRegexString}`);
 /** Main Lexer class */
 export default class Lexer implements LexerInterface {
     input: string;
-    pos: number;
+    tokenRegex: RegExp;
 
     constructor(input: string) {
         // Separate accents from characters
         this.input = input;
-        this.pos = 0;
+        this.tokenRegex = new RegExp(tokenRegexString, 'g');
     }
 
     /**
      * This function lexes a single token.
      */
     lex(): Token {
         const input = this.input;
-        const pos = this.pos;
+        const pos = this.tokenRegex.lastIndex;
         if (pos === input.length) {
             return new Token("EOF", new SourceLocation(this, pos, pos));
         }
-        const match = matchAt(tokenRegex, input, pos);
+        const match = this.tokenRegex.exec(input);
```

Comment:
I can think of one important difference here: if there's something that doesn't match the regex, `exec` will (I think) skip over it, while `matchAt` will fail.  Thoughts on how to fix this?

Reply:
@edemaine `matchAt` and `exec` both return `null` when match fails, and a `ParseError` will be thrown below.

Reply:
If there's a match that occurs after part of the string which isn't matched by the regex, `exec` will happily jump to the next match and return that.  We could keep track of the current position and then verify that match that `exec` occurs at that position.

Reply:
@kevinbarabash Ah, I didn't know, hence the name match _At_. Added an assertion to check match index matches the last index. Also updated a test case to test regression.